MOF 829 - Add Examples for Form API (#1)
moffin-tech/api-examples: request-forms/README.md

While we provide these Forms through our webapp, you can also make requests
to our API to create forms for users, validate their email/phone and submit them to 
our platform, allowing you to create your own webapp and integrate it with our services.

## Using the Forms API

### Create your moffin account
First you will need an account in the Moffin Sandbox
environment. You can create one easily in this
[page](https://moffin.mx/sign_up).

### Generate an Access Token
Go to the configuration panel in moffin to create an
*Access Token*.

Go to "Configuración" > "API" > "Crear nuevo token":
* [Configure token in sandbox](https://sandbox.moffin.mx/configuracion/api)
* [Configure token in production](https://app.moffin.mx/configuracion/api)

Then you can configure the token as is show in this example.
```js
const client = Axios.create({
  baseURL: 'https://sandbox.moffin.mx/api/v1',
  headers: {

>   'Authorization': 'Token ' + process.env.API_TOKEN

Reviewer: ```suggestion
    'Authorization': 'Token ' + process.env.API_TOKEN
```

### Create a form
Create a new form using the provided data, keep in mind this doesn't actually submit the form or verify its data. Keep track of the Form ID for the rest of the endpoint calls.

You'll need a Form Config ID, which establishes the format used for the form and some of its configuration. You'll find the Form Config ID for each form defined in your account, but for the following examples and testing in our sandbox environment, you can just use the id `1`.

Please contact support if you need help finding an appropiate Form Config ID for your forms.

```js
  const { data: form } = await client.post('/form/1', {
    accountType: 'PF',
    loanAmount: '10000',
    email: 'test-dev@moffin.mx',
    firstName: 'Prueba',
    phone: '+52 <your phone number>', // Use your own phone number if you wish to test the send-sms endpoint later
    rfc: 'PRPU800101111',
    curp: 'PETJ970221HJCRRN09',
    birthdate: '1997-01-01',
    address: 'Jaime Balmes',
    city: 'Cd de Mexico',
    state: 'CMX',
    zipCode: '11510',
    exteriorNumber: '8',
    neighborhood: 'Los morales',
    country: 'MX',
    nationality: 'MX'
  })
  console.log('Created Form ID', form.formId)
```

> If the Form is created without issues, you'll get the form ID back, we will use that ID for the remaining requests.

Reviewer: I would add an example:
```
```suggestion
Once the Form is successfully create you'll get the form ID (`form.id`) back, we will use that ID for the remaining requests.
```

Author: Te refieres a un ejemplo del response? Tecnicamente esta justo en las lineas de arriba

Author: Igual añadi un snippet del response

Reviewer: Si, esa es la idea. Entonces todo bien

```json5
// SMS Response Body 
{
  timeUntilNewAttempt: '2022-10-12T17:41:56.185Z',
  remainingAttempts: 2,
  sent: true
}
```

```json5
// Email Response Body 
{
  timeUntilNewAttempt: '2022-10-12T17:41:56.185Z',
  remainingAttempts: 2,
  sent: true
}
```

### Verify NIP
You can use this endpoint to verify the NIP sent by the previous endpoints, keep in mind this endpoint will return 204 if the verification is succesful, not 200.

```js
 await client.post(`/form/nip/validate/${form.formId}`, {
    nip: code // code is the NIP received by the user
 })
```

### Submit the Form
Once the form has been populated with the correct information, and the it has been verified, you can submit the form to our system using this endpoint. Make sure to request all the required fields when creating the form (provided in the Create Form example).

```js
 await client.post(`/form/submit/${form.formId}`, {
    validateNIPAuthentication: true // Ensure the NIP was validated
 })

> ```

Reviewer: También agregaría un comentario explicando para que es el campo `validateNIPAuthentication` y cuando usarlo.

Also así:

_If you have built your own solution to validate NIP outside Moffin, you can set the flag `validateNIPAuthentication` as `false` to disable the validation. If you set this flag as true, we will check if at least the phone number or email has been validated before the form submission, if none is validate we will return an error `403`_
